pkg/distributor/distributor.go

```diff
@@ -432,21 +432,25 @@ func shardStream(stream logproto.Stream, cfg Config, streamSharder StreamSharder
 
 	entriesPerWindow := float64(len(stream.Entries)) / float64(shards) // divide and keep decimal value.
 	for i := 0; i < shards; i++ {
+		fIdx := float64(i)
+		lowerBound := int(fIdx * entriesPerWindow)
+		upperBound := min(int(entriesPerWindow*(1+fIdx)), len(stream.Entries))
+		if lowerBound > upperBound {
+			continue
```

Comment:
It _may_ be worth logging a debug entry here, or incrementing a metric - otherwise your shard splitting may be silently failing with no real indication why

Reply:
for this first iteration I think it makes sense to have this kind of debugging, so I added a log line here reporting the scenario. WDYT?

Reply:
:+1: perfect, thanks